### DISCUSSION TITLE
While running extension, it opens a new window, how to disable and make running on current folder

### DISCUSSION THREAD

**Discussion by Kjpraveen26** [microsoft/vscode-discussions#1362] [Extension Development QnA] [answered]
While running extension, it opens a new window, how to disable and make running on current folder or how to give the folder to run extention on that folder.

**Comment by theReynald** [comment]
Hi Kjpraveen26.

I asked copilot your question and in provided the following. Let me know if this helps.

**_Copilot suggestions_**

When developing a VS Code extension, you can control whether the extension runs in the current window or a new window by configuring the launch.json file in your extension's .vscode directory. This file is used to define how the extension is launched during development.

To ensure that your extension runs in the current window or specify a folder to run the extension on, you can modify the launch.json file as follows:

1. Open the launch.json file: If it doesn't exist, create it in the .vscode directory of your extension project.
2. Configure the launch.json file: Add or modify the configurations to specify the folder to run the extension on.

Here is an example of how to configure launch.json to run the extension in the current window or specify a folder:
```json
{
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Run Extension",
            "type": "extensionHost",
            "request": "launch",
            "runtimeExecutable": "${execPath}",
            "args": [
                "--extensionDevelopmentPath=${workspaceFolder}",
                "${workspaceFolder}" // This ensures the extension runs in the current folder
            ],
            "outFiles": [
                "${workspaceFolder}/out/**/*.js"
            ],
            "preLaunchTask": "npm: watch"
        },
        {
            "name": "Run Extension on Specific Folder",
            "type": "extensionHost",
            "request": "launch",
            "runtimeExecutable": "${execPath}",
            "args": [
                "--extensionDevelopmentPath=${workspaceFolder}",
                "/path/to/specific/folder" // Replace with the path to the folder you want to run the extension on
            ],
            "outFiles": [
                "${workspaceFolder}/out/**/*.js"
            ],
            "preLaunchTask": "npm: watch"
        }
    ]
}
```

**Explanation:**

- "type": "extensionHost": Specifies that this configuration is for running an extension.
- "request": "launch": Indicates that this configuration is for launching the extension.
- "runtimeExecutable": "${execPath}": Uses the current VS Code executable.
- "args":

  - --extensionDevelopmentPath=${workspaceFolder}: Specifies the path to the extension's development folder.
  - ${workspaceFolder}: Ensures the extension runs in the current folder.
  - "/path/to/specific/folder": Replace with the path to the specific folder you want to run the extension on.

- "outFiles": Specifies the output files for debugging.
- "preLaunchTask": Specifies a task to run before launching the extension (e.g., npm: watch).

**Running the Extension:**

- To run the extension in the current folder, select the "Run Extension" configuration from the Run and Debug view in VS Code and start debugging.
- To run the extension on a specific folder, select the "Run Extension on Specific Folder" configuration and start debugging.

This setup allows you to control whether the extension runs in the current window or on a specified folder during development.

**Comment by olguzzar** [comment, marked as answer]
Hi @Kjpraveen26, did using the above launch.json args help? 

Alternatively, if your goal is to have a certain folder open when testing your extension, in the new VS Code window that opens with your extension running, you can simply open that folder. The VS Code window running the Extension Development Host can be treated just like any other VS Code window, so you can open, edit, save files just like you normally would.

If neither of these are solving the issue you are having, please let us know!!
